darc/darc.go

```diff
@@ -1180,6 +1191,103 @@ func parseIDEvmContract(in string) (Identity, error) {
 	}, nil
 }
 
+func evalThreshold(visited map[string]bool, getDarc GetDarc,
+	attrFuncs AttrInterpreters, acceptDarc bool, s string, ids []string) error {
+
+	// we have something of form 'threshold<1/2,darc:aa,ed25519:bb,...>'
+
+	s = strings.TrimPrefix(s, "threshold<")
+	s = strings.TrimSuffix(s, ">")
+
+	entries := strings.Split(s, ",")
+
+	if len(entries) < 1 {
+		return xerrors.Errorf("expected at least the threshold fraction: %s", s)
+	}
+
+	fractionStr := entries[0]
+
+	ok := regexp.MustCompile(`\d+/\d+`).MatchString(fractionStr)
+	if !ok {
+		return xerrors.Errorf("the threshold fraction is incorrect: %s", fractionStr)
+	}
+
+	fractionElem := strings.Split(fractionStr, "/")
```

Comment:
You can simplify it by using a single regex, which would be safer IMO. smth like `threshold<(\d+)/(\d+)(,((darc:[[:xdigit:]]+)|(ed25519:[[:xdigit:]]+)))+>`

Reply:
Hum, you're right, maybe use a simpler `threshold<(\d+)/(\d+)(,[^,]+)+>` at least? It allows to extract the numerator & denominator, and each id.

Reply:
Not working: https://play.golang.org/p/oPV3LrRjNRG

Reply:
@tharvik I think the test for 0 is good, but the denominator does not need to be the same as the number of entries. I also started with the idea of only giving the number of signatures needed, so getting rid of the denominator. But then you need to update the number every time you add/remove a signature.

So I prefer @nkcr 's solution where you can give a fraction that will stay the same. Even though, to be perfect, it should allow to do things like `(- 1, 2 / 3)` to use it in a BFT context where you need `2f+1` signatures... But that's definitely out of scope for this PR.